docs/adr/0008-allowing-more-types-as-errors.md

```diff
@@ -0,0 +1,139 @@
+# Allowing more types to be used as errors.
+
+* Status: proposed
+* Deciders: Hopefully consensus on #1613
+
+## Current UniFFI error handling
+
+UniFFI functions currently must report error returns via an `Enum` - ie, the `E` in `Result<T, E>` must be an Enum.
+
+This ADR proposes moving towards a model where any type could be an `E`, and specfically, proposes allowing any uniffi "interface"
```

Comment:
Maybe `s/any type could/more types can/`.  I don't think we ever want ints or floats to be an error type.